infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -197,13 +197,22 @@ private boolean isSnapshotApplyPending(LogReplicationMetadataManager metadataMgr
      */
     protected synchronized boolean isLeader(@Nonnull RequestMsg request,
                                             @Nonnull ChannelHandlerContext ctx,
-                                            @Nonnull IServerRouter router) {
+                                            @Nonnull IServerRouter router, boolean isLogEntry) {
         // If the current cluster has switched to the active role (no longer the receiver) or it is no longer the leader,
         // skip message processing (drop received message) and nack on leadership (loss of leadership)
         // This will re-trigger leadership discovery on the sender.
         boolean lostLeadership = isActive.get() || !isLeader.get();
 
         if (lostLeadership) {
+
+            if (isLogEntry) {
+                LogReplicationEntryMsg entryMsg = request.getPayload().getLrEntry();
+                LogReplicationEntryType entryType = entryMsg.getMetadata().getEntryType();
+                log.warn("Received message of type {} while NOT LEADER. snapshotSyncSeqNumber={}, ts={}, syncRequestId={}", entryType,
```

Comment:
nit - ..while not leader or in active state...

Reply:
Not sure what you mean by or in active state, this is the standby rejecting a message as it is not the leader.

Reply:
I meant, lostLeadersip can be true if node is in active cluster or standby but not the leader.